### PR TITLE
Frontend dynamic credentials update

### PR DESCRIPTION
**Pull request by JKonradoslaw**
Created this branch to add the functionality for the front, to dynamically change the password, instead of making a new Job file.

Currently is only for the jobs that were already run.

I will add this functionality to jobs that also were not run yet and add the possibility to change the login dynamically too!

### CODE REVIEW

File: backend/backend.py [hunk 1]
```diff
@@ -13,10 +13,11 @@
 @click.option("-t", "--thresholds-file", default="DefaultThresholds")
 @click.option("-d", "--debug", is_flag=True)
 @click.option("-c", "--concurrent-connections", type=int)
+@click.option("-p", "--password-dynamically", default=None)
```
Comment (bhjelmar) [inline backend/backend.py:16]: Let's change the variable name from password-dynamically to password. To let people know it's dynamic, we can add a help text https://click.palletsprojects.com/en/8.1.x/documentation/

File: backend/core/Engine.py [hunk 2]
```diff
@@ -82,20 +82,29 @@ def __init__(self, jobFileName: str, thresholdsFileName: str, concurrentConnecti
             concurrentConnections = 50 if concurrentConnections is None else concurrentConnections
         AsyncioUtils.init(concurrentConnections)
 
+        self.password_dynamically = password_dynamically
```
Comment (bhjelmar) [inline backend/core/Engine.py:85]: We don't need to save the password to the self context, I think this line can be removed

File: backend/core/Engine.py [hunk 3]
```diff
@@ -82,20 +82,29 @@ def __init__(self, jobFileName: str, thresholdsFileName: str, concurrentConnecti
...
         self.controllers = [
             AppDService(
                 host=controller["host"],
                 port=controller["port"],
                 ssl=controller["ssl"],
-                account=controller["account"],
-                username=controller["username"],
-                pwd=controller["pwd"],
-                verifySsl=controller.get("verifySsl", True),
+                account=controller["account"],  # Attention, a user can pass multiple controllers,
```
Comment (bhjelmar) [inline backend/core/Engine.py:93]: Let's remove some of these comments before the PR is accepted, fine for now tho

File: backend/core/Engine.py [hunk 4]
```diff
@@ -82,20 +82,29 @@ def __init__(self, jobFileName: str, thresholdsFileName: str, concurrentConnecti
...
+                username=controller["username"],  # i.e. multiple controller credentials in one job.file
+                pwd=password_dynamically if password_dynamically != "None" else controller["pwd"],  # This line
+                verifySsl=controller.get("verifySsl", True),  # will set passwords in ALL controllers in job.file
                 useProxy=controller.get("useProxy", False),
             )
+            # Put an information, taht if the user puts the pwd dynamicly
```
Comment (bhjelmar) [inline backend/core/Engine.py:99]: Remove comment before PR accepted

File: frontend/views/jobPreviouslyExecuted.py [hunk 5]
```diff
@@ -12,7 +12,9 @@
 from utils.docker_utils import runConfigAssessmentTool, isDocker
 
 
-def jobPreviouslyExecuted(client: APIClient, jobName: str, debug: bool, concurrentConnections: int, platformStr: str, tag: str):
+def jobPreviouslyExecuted(
+    client: APIClient, jobName: str, debug: bool, concurrentConnections: int, password_dynamically: str, platformStr: str, tag: str
```
Comment (bhjelmar) [inline frontend/views/jobPreviouslyExecuted.py:16]: Same as above: I think propagating this variable down is needless? It is set in the jobPreviously/NotYetExecuted elements.

File: backend/core/Engine.py [hunk 6]
```diff
@@ -82,20 +82,29 @@ def __init__(self, jobFileName: str, thresholdsFileName: str, concurrentConnecti
...
                 host=controller["host"],
                 port=controller["port"],
                 ssl=controller["ssl"],
-                account=controller["account"],
-                username=controller["username"],
-                pwd=controller["pwd"],
-                verifySsl=controller.get("verifySsl", True),
+                account=controller["account"],  # Attention, a user can pass multiple controllers,
+                username=controller["username"],  # i.e. multiple controller credentials in one job.file
+                pwd=password_dynamically if password_dynamically != "None" else controller["pwd"],  # This line
```
[the remark this answers is not in the document]
Comment (JKonradoslaw) [reply backend/core/Engine.py:95]: Litteraly None

**Pull request closed without merging**